server/src/main/java/com/linecorp/centraldogma/server/storage/encryption/DefaultEncryptionStorageManager.java

```diff
@@ -0,0 +1,418 @@
+        options = new DBOptions();
+        try {
+            final List<ColumnFamilyHandle> columnFamilyHandles = new ArrayList<>();
+            rocksDb = RocksDB.open(options, rocksDbPath, columnFamilyDescriptors, columnFamilyHandles);
+            ColumnFamilyHandle defaultColumnFamilyHandle = null;
+            ColumnFamilyHandle metadataColumnFamilyHandle = null;
+            for (ColumnFamilyHandle handle : columnFamilyHandles) {
+                if (Arrays.equals(RocksDB.DEFAULT_COLUMN_FAMILY, handle.getName())) {
+                    defaultColumnFamilyHandle = handle;
+                } else if (Arrays.equals(ENCRYPTION_METADATA_COLUMN_FAMILY.getBytes(StandardCharsets.UTF_8),
+                                         handle.getName())) {
+                    metadataColumnFamilyHandle = handle;
+                }
+            }
+            if (defaultColumnFamilyHandle == null || metadataColumnFamilyHandle == null) {
+                throw new EncryptionStorageException("Failed to get required column family handles.");
+            }
+            this.defaultColumnFamilyHandle = defaultColumnFamilyHandle;
+            this.metadataColumnFamilyHandle = metadataColumnFamilyHandle;
+        } catch (RocksDBException e) {
+            throw new EncryptionStorageException("Failed to open RocksDB", e);
+        }
+    }
+
+    @Override
+    public boolean enabled() {
+        return true;
+    }
+
+    @Override
+    public CompletableFuture<byte[]> generateWdek() {
+        final byte[] dek = EncryptionKeyUtil.generateAes256Key();
+        return keyManagementService.wrapDek(dek);
+    }
+
+    @Override
+    public SecretKey getDek(String projectName, String repoName) {
+        requireNonNull(projectName, "projectName");
+        requireNonNull(repoName, "repoName");
+        final byte[] wdek;
+        try {
+            wdek = rocksDb.get(repoWdekDbKey(projectName, repoName).getBytes(StandardCharsets.UTF_8));
+        } catch (RocksDBException e) {
+            throw new EncryptionStorageException(
+                    "Failed to get WDEK of " + projectRepo(projectName, repoName), e);
+        }
+
+        if (wdek == null) {
+            throw new EncryptionStorageException(
+                    "WDEK of " + projectRepo(projectName, repoName) + " does not exist");
+        }
+
+        try {
+            return new SecretKeySpec(keyManagementService.unwrapWdek(wdek).get(10, TimeUnit.SECONDS), "AES");
+        } catch (Throwable t) {
+            throw new EncryptionStorageException(
+                    "Failed to unwrap WDEK of " + projectRepo(projectName, repoName), t);
+        }
+    }
+
+    @Override
+    public void storeWdek(String projectName, String repoName, byte[] wdek) {
+        requireNonNull(projectName, "projectName");
+        requireNonNull(repoName, "repoName");
+        requireNonNull(wdek, "wdek");
+        final byte[] wdekKeyBytes = repoWdekDbKey(projectName, repoName).getBytes(StandardCharsets.UTF_8);
```

Comment:
Question) Just to make sure, is the wdek stored in the default column family intentionally?

I guess this is fine since encrypted ids probably don't collide with the key, but just wanted to make sure it is intentional the wdek is stored together with user data.